### PR TITLE
Use the tmp directory when installing homebrew. Fixes #18

### LINKED ISSUE #18: Unable to use this role if /usr/local is not empty
I encountered this problem when I was trying to run `mac-dev-playbook`. Before running the playbook, I installed ansible using pip. This role fails to clone the homebrew repo if /usr/local is not empty.

As a workaround, I installed homebrew manually, then re-ran the playbook.

### PR DESCRIPTION
**Pull request by andrewkboyd**
This resolves #18 for me, and hopefully others.

This is the first time I have used ansible, so feel free to suggest how I could clean it up and I will make the changes as quickly as I can.  This is working for me on El Cap.

### CODE REVIEW

**Comment by ksylvan** [comment]
Looks to me (on El Capitan) that the easy way is to run the homebrew install script (for the homebrew homepage) when installing brew. That way we don't have to git clone to /tmp and then copy it to /usr/local. It works for me.

In fact, this playbook works in El Capitan:

```
---
- hosts: all
  gather_facts: false
  tasks:
    - name: Determine if brew binary exists
      stat: path=/usr/local/bin/brew
      register: brew_binary
      changed_when: false

- hosts: all
  gather_facts: false
  tasks:
    - name: install homebrew
      shell: >
        /usr/bin/ruby -e "$(curl -fsSL
        https://raw.githubusercontent.com/Homebrew/install/master/install)"
        < /dev/null
      when: brew_binary.stat.exists == false
```

**Comment by geerlingguy** [comment]
@ksylvan - I haven't had time to review the PR yet, but I know I'm not a fan of using any kind of 'curl install wrapper' in my roles; I like to expose any of the installation 'magic' in the role itself. It does make things _slightly_ more annoying from a maintenance perspective... but I don't like the idea of using curl to pull any kind of malleable script that runs on my computer.

Basically:

  - `-1 to curl | bash` (or ruby, etc.)
  - `-∞ to curl | sudo bash`

**Comment by ksylvan** [comment]
Understood. Then perhaps we should look at the install script and learn why/how it succeeds. It probably does all the same things under the hood (like chown /usr/local, etc.)

File: tasks/main.yml [hunk 1]
```diff
@@ -13,14 +13,24 @@
...
+    dest: /tmp/brew
     version: master
-    dest: "{{ homebrew_install_path }}"
     update: no
     accept_hostkey: yes
     depth: 1
+  when: homebrew_binary.stat.islnk is not defined
+
+- name: Move homebrew to install directory
+  synchronize:
```
Comment (geerlingguy) [inline tasks/main.yml]: Could we maybe use `command: mv /tmp/brew /usr/local` instead? Or is there another reason the `synchronize` module is better? I would just rather create a duplicate, even though the first would be cleaned up at a later time...

**Comment by geerlingguy** [comment]
Closing in favor of #35; we will need to do a little more refactoring overall.

**Pull request closed without merging**